devtools/x/src/generate_summaries.rs

```diff
@@ -77,25 +98,39 @@ fn write_summary(
     feature_query: FeatureQuery<'_>,
     cargo_opts: &CargoOptions<'_>,
     out_dir: &Path,
+    output_format: Option<&str>,
 ) -> crate::Result<()> {
     let build_set = feature_query.resolve_cargo(cargo_opts)?;
-
-    let mut out = format!(
-        "# Summary for Diem subset '{}'. @generated by x.\n\
-         # To regenerate, run 'cargo x generate-summaries'.\n\n",
-        name
-    );
-    build_set
-        .to_summary(cargo_opts)?
-        .write_to_string(&mut out)
-        .with_context(|| format!("error while generating summary for '{}'", name))?;
-
-    let summary_path = out_dir.join(format!("summary-{}.toml", name));
-
-    fs::write(&summary_path, &out).with_context(|| {
-        format!(
-            "error while writing summary file {}",
-            summary_path.display()
-        )
-    })
+    let summary = build_set.to_summary(cargo_opts)?;
+
+    let (out, path) = match output_format {
+        Some("json") => {
+            let out = serde_json::to_string(&summary)?;
+            let summary_path = out_dir.join(format!("summary-{}.json", name));
+            (out, summary_path)
+        }
+        Some("toml") => {
+            let out = toml::to_string(&summary)?;
+            let summary_path = out_dir.join(format!("summary-{}.toml", name));
+            (out, summary_path)
```

Comment:
Please continue to use the `write_to_string` method.

Reply:
you mean to have the text at the beginning of the doc? The problem with that is that it makes the json file invalid, but it works with toml so I can have it for toml if that's what you're asking?

Reply:
oh wait, the default text was already toml 🤦🏻‍♂️

Reply:
done!